### PR TITLE
Simplify `startEvaluator()`

### PR DESCRIPTION
**Pull request by tolikzinovyev**
## Summary

The current code creates an impression that the evaluator can be used for round 0 which is not true. This PR simplifies the code, so this is more evident.

## Test Plan

Relying on existing tests.

### CODE REVIEW

File: ledger/eval.go [hunk 1]
```diff
@@ -383,20 +383,32 @@ func (l *Ledger) StartEvaluator(hdr bookkeeping.BlockHeader, paysetHint int) (*B
 }
 
 func startEvaluator(l ledgerForEvaluator, hdr bookkeeping.BlockHeader, proto config.ConsensusParams, paysetHint int, validate bool, generate bool) (*BlockEvaluator, error) {
+	if hdr.Round == 0 {
+		return nil, fmt.Errorf("cannot start evaluator for round 0")
```
Comment (tsachiherman) [inline ledger/eval.go:387]: create global constant error using errors.New(), and use it here.

File: ledger/eval.go [hunk 2]
```diff
@@ -412,35 +424,23 @@ func startEvaluator(l ledgerForEvaluator, hdr bookkeeping.BlockHeader, proto con
...
-
-		var ok bool
-		prevProto, ok = config.Consensus[eval.prevHeader.CurrentProtocol]
-		if !ok {
-			return nil, protocol.Error(eval.prevHeader.CurrentProtocol)
-		}
+	prevProto, ok := config.Consensus[prevHeader.CurrentProtocol]
+	if !ok {
+		return nil, protocol.Error(prevHeader.CurrentProtocol)
+	}
```
Comment (tsachiherman) [inline ledger/eval.go:432]: move this section right after 
```golang
prevHeader, err := l.BlockHdr(hdr.Round - 1)
 	if err != nil {
 		return nil, fmt.Errorf(
 			"can't evaluate block %v without previous header: %v", hdr.Round, err)
 	}
```
for further clarity.

**Review by tsachiherman — CHANGES_REQUESTED** [review]
few small requests, but looks good otherwise.

**Comment by algorandskiy** [comment]
I personally do not see any value in this change.

**Review by winder — COMMENTED** [review]
Are we sure this function is never called on round 0? It appears that previously it would be a NoOp (or a crash?), now it's an error.

**Comment by tolikzinovyev** [comment]
I tried using it for Indexer for round 0, I got an error. It's probably in
```
	poolAddr := eval.prevHeader.RewardsPool
	// get the reward pool account data without any rewards
	incentivePoolData, _, err := l.LookupWithoutRewards(eval.prevHeader.Round, poolAddr)
	if err != nil {
		return nil, err
	}
```

`eval.prevHeader` is unset, so we are trying to get a balance for zero address.

**Review by tsachiherman — APPROVED** [review]
look good. thanks for the changes.

**Pull request merged**